src/types.js

```diff
@@ -32,8 +19,40 @@ export interface IPositionLog extends ILog {
     +info: (message: string, line?: string | number) => void;
     +log: (message: string, line?: string | number) => void;
     +warn: (message: string, line?: string | number) => void;
+    +mismatch: (message: string, line?: string | number) => void;
+    +fix: (message: string, line?: string | number) => void;
 }
 
+export type FixAction =
+    | {
+          type: "delete",
+          line: number,
+          description: string,
+          // The original declaration being deleted.
+          declaration: string,
+      }
+    | {
+          type: "replace",
+          line: number,
```

Comment:
A fix is always just a single line. It would be nice, perhaps, to support moving a line from one place to another (such as would be needed to auto-fix a "target after content start" error), but that's pretty complicated in our current world (that said, it could be achieved by adding an "insert" type of fix and combining it with a "delete"...something to think about).